python/ray/experimental/collective/reducescatter.py

```diff
@@ -0,0 +1,92 @@
+import logging
+from typing import List, Optional, Union
+
+import ray
+from ray.dag.collective_node import CollectiveOutputNode, _CollectiveOperation
+from ray.dag.constants import (
+    BIND_INDEX_KEY,
+    COLLECTIVE_OPERATION_KEY,
+    PARENT_CLASS_NODE_KEY,
+)
+from ray.experimental.channel.torch_tensor_type import Communicator, TorchTensorType
+from ray.experimental.util.types import ReduceOp
```

Comment:
Create two new enums `AllReduceOp` and `ReduceScatterOp` inherited from `ReduceOp`. They do not need to introduce new enum values, just use `pass`. Such that we don't need a str to tell whether this is a allreduce or reducescatter, simply just check if their types are `AllReduceOp` and `ReduceScatterOp`.